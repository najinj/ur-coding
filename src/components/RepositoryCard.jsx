/* eslint-disable camelcase */
import React from "react";
import styled from "styled-components";
import { string, number, shape } from "prop-types";

const Container = styled.div`
  display: flex;
`;
const Image = styled.img`
  width: 200px;
`;
const Delails = styled.div`
  width: 100%;
  border: 1px solid;
`;

const RepositoryCard = ({
  owner,
  name,
  description,
  stargazers_count,
  open_issues_count,
  updated_at
}) => {
  return (
    <Container className="repo-container">
      <div className="avatar-container">
        <Image src={owner.avatar_url} alt="avatar_url" />
      </div>
      <Delails className="repo-details">
        <h5>{name}</h5>
        <p>{description}</p>
        <div className="repo-rating">
          <div className="repo-stars">
            <span>Stars : {stargazers_count}</span>
          </div>
          <div className="repo-issues">
            <span>Issues : {open_issues_count}</span>
          </div>
          <div className="repo-last-update">
            <span>Submited {updated_at} ago by </span>
          </div>
        </div>
      </Delails>
    </Container>
  );
};

RepositoryCard.propTypes = {
  name: string.isRequired,
  owner: shape(),
  open_issues_count: number,
  description: string,
  stargazers_count: number,
  updated_at: string
};

RepositoryCard.defaultProps = {
  owner: {},
  description: "",
  stargazers_count: 0,
  open_issues_count: 0,
  updated_at: ""
  // I know the github API always sends this , but this just to show case well whatever xD
};

export default RepositoryCard;
