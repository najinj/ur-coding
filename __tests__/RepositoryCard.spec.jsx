import React from "react";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import apiData from "./data";
import RepositoryCard from "../src/components/RepositoryCard";

configure({ adapter: new Adapter() });

test("RepositoryCard renders correctly", () => {
  const repo = apiData.items[0];
  // eslint-disable-next-line react/jsx-props-no-spreading
  const component = shallow(<RepositoryCard {...repo} />);
  expect(toJson(component)).toMatchSnapshot();
});
