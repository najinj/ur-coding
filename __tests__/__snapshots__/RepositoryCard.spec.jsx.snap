// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryCard renders correctly 1`] = `
<styled.div
  className="repo-container"
>
  <div
    className="avatar-container"
  >
    <styled.img
      alt="avatar_url"
      src="https://avatars1.githubusercontent.com/u/1342004?v=4"
    />
  </div>
  <styled.div
    className="repo-details"
  >
    <h5>
      eng-practices
    </h5>
    <p>
      Google's Engineering Practices documentation
    </p>
    <div
      className="repo-rating"
    >
      <div
        className="repo-stars"
      >
        <span>
          Stars : 
          6914
        </span>
      </div>
      <div
        className="repo-issues"
      >
        <span>
          Issues : 
          4
        </span>
      </div>
      <div
        className="repo-last-update"
      >
        <span>
          Submited 
          2019-09-13T14:01:39Z
           ago by 
        </span>
      </div>
    </div>
  </styled.div>
</styled.div>
`;
